File: LeetCode/Graph/BFS/Python/Leetcode1302-deepest-leaves-sum.py
class Solution:
    def deepestLeavesSum(self, root: Optional[TreeNode]) -> int:
        q,ans = deque([root]),0
        while q:
            n = len(q)
            ans = 0
            for i in range(n):
                t = q.popleft()
                ans += t.val
                if t.left: q.append(t.left)
                if t.right: q.append(t.right)
        return ans

File: LeetCode/Graph/BFS/Golang/Leetcode1302-deepest-leaves-sum.go
package sbgo

func deepestLeavesSum(root *TreeNode) int {
	ans, q := 0, []*TreeNode{root}
	for len(q) != 0 {
		n := len(q)
		ans = 0
		for i := 0; i < n; i++ {
			t := q[0]
			ans += t.Val
			q = q[1:]
			if t.Left != nil {
				q = append(q, t.Left)
			}
			if t.Right != nil {
				q = append(q, t.Right)
			}
		}
	}
	return ans
}
